casper/src/test/scala/coop/rchain/casper/genesis/contracts/RevIssuanceTest.scala

```diff
@@ -29,107 +32,106 @@ class RevIssuanceTest extends FlatSpec with Matchers {
-      val transferResult  = getDataUnsafe(runtimeManager, postTransferHash, destination)
-      assert(transferSuccess.head.exprs.head.getGString == "Success") //assert transfer success
-      assert(transferResult.nonEmpty)
-
-      activeRuntime.close()
-    }
+        )
+        for {
+          runtimeManager        <- RuntimeManager.fromRuntime(activeRuntime)
+          emptyHash             = runtimeManager.emptyStateHash
+          postGenState          <- runtimeManager.computeState(emptyHash)(genesisDeploys)
+          (postGenHash, _)      = postGenState
+          postDeployState       <- runtimeManager.computeState(postGenHash)(transferDeployData :: Nil)
+          (postTransferHash, _) = postDeployState
+          transferSuccess <- runtimeManager.getData(postTransferHash)(
+                              Par().copy(exprs = Seq(Expr(GString(transferStatusOut))))
+                            )
+          _ = assert(transferSuccess.head.exprs.head.getGInt == 15) //assert transfer success
+        } yield ()
+      }
+      .runSyncUnsafe(60.seconds)
   }
 }
 
 object RevIssuanceTest {
-  def preWalletUnlockDeploy[F[_]: Concurrent](
-      ethAddress: String,
-      pubKey: String,
-      secKey: Array[Byte],
-      statusOut: String
-  )(implicit runtimeManager: RuntimeManager[F]): F[DeployData] =
-    BondingUtil.preWalletUnlockDeploy[F](ethAddress, pubKey, secKey, statusOut).map { code =>
-      ConstructDeploy.sourceDeploy(
-        code,
-        System.currentTimeMillis(),
-        accounting.MAX_VALUE
-      )
-    }
+  private def walletTransfer(
+      from: RevAddress,
+      amount: Long,
+      destination: RevAddress,
+      transferStatusOut: String,
+      secKey: Array[Byte]
+  ): DeployData = {
+    val transferDeployCode =
+      s"""
+         |new rl(`rho:registry:lookup`), revVaultCh, vaultCh, revVaultKeyCh, stdout(`rho:io:stdout`), resultCh, destinationCh, balanceCh in {
+         |  rl!(`rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z`, *revVaultCh) |
+         |  for(@(_, RevVault) <- revVaultCh) {
+         |    @RevVault!("findOrCreate", "${from.toBase58}", *vaultCh) |
+         |    @RevVault!("deployerAuthKey", *revVaultKeyCh) |
+         |    for (@(true, vault) <- vaultCh; key <- revVaultKeyCh) {
+         |      @RevVault!("findOrCreate", "${destination.toBase58}", *destinationCh) | // Creates vault in case it doesn't exist
+         |      @vault!("transfer", "${destination.toBase58}", $amount, *key, *resultCh) |
+         |      for (@(true, Nil) <- resultCh) {
+         |        for (@(true, destinationVault) <- destinationCh) {
+         |          @destinationVault!("balance", *balanceCh) |
+         |          for (@balance <- balanceCh) {
+         |            @"$transferStatusOut"!(balance)
+         |          }
+         |        }
+         |      }
+         |    }
+         |  }
+         |}""".stripMargin
```

Comment:
In the light of the yesterday discussions about moving non-trivial Rholang code to resource files, I think this qualifies as non-trivial.

I would move create RevIssuanceTest/Transfer.rhox with content:
```

new rl(`rho:registry:lookup`), revVaultCh, vaultCh, revVaultKeyCh, stdout(`rho:io:stdout`), resultCh, destinationCh, balanceCh in {
  rl!(`rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z`, *revVaultCh) |
  for(@(_, RevVault) <- revVaultCh) {
  @RevVault!("findOrCreate", "$$from$$", *vaultCh) |
    @RevVault!("deployerAuthKey", *revVaultKeyCh) |
    for (@(true, vault) <- vaultCh; key <- revVaultKeyCh) {
      @RevVault!("findOrCreate", "${destination.toBase58}", *destinationCh) | // Creates vault in case it doesn't exist
      @vault!("transfer", "$$to$$", $$amount$$, *key, *resultCh) |
      for (@(true, Nil) <- resultCh) {
        for (@(true, destinationVault) <- destinationCh) {
          @destinationVault!("balance", *balanceCh) |
          for (@balance <- balanceCh) {
            @"$$transferStatusOut$$"!(balance)
          }
        }
      }
    }
  }
}

```

...and the code here would look like
```val transferDeployCode = CompiledRholangTemplate("RevIssuanceTest/Transfer.rhox", 
   "from" -> from.toBase58,
   "to" -> destination.toBase58,
   "amount" -> amount,
   "transferStatusOut" -> transferStatusOut
).code```

Reply:
@odeac see my comment above

Reply:
I agree that, if this can be written as a RhoSpec test, it would be better.